File: Python/OneLang/One/Transforms/InferTypesPlugins/Helpers/InferTypesPlugin.py
from OneLangStdLib import *
import OneLang.One.ErrorManager as errorMan
import OneLang.One.Ast.Expressions as exprs
import OneLang.One.Transforms.InferTypes as inferTypes
import OneLang.One.Ast.Statements as stats
import OneLang.One.Ast.Types as types

class InferTypesPlugin:
    def __init__(self, name):
        self.main = None
        self.error_man = None
        self.name = name
    
    def can_transform(self, expr):
        return False
    
    def can_detect_type(self, expr):
        return False
    
    def transform(self, expr):
        return expr
    
    def detect_type(self, expr):
        return False
    
    def handle_property(self, prop):
        return False
    
    def handle_lambda(self, lambda_):
        return False
    
    def handle_method(self, method):
        return False
    
    def handle_statement(self, stmt):
        return False

File: PHP/src/OneLang/One/Transforms/InferTypesPlugins/Helpers/InferTypesPlugin.php
<?php

namespace OneLang\One\Transforms\InferTypesPlugins\Helpers\InferTypesPlugin;

use OneLang\One\ErrorManager\ErrorManager;
use OneLang\One\Ast\Expressions\Expression;
use OneLang\One\Transforms\InferTypes\InferTypes;
use OneLang\One\Ast\Statements\Statement;
use OneLang\One\Ast\Types\Property;
use OneLang\One\Ast\Types\Lambda;
use OneLang\One\Ast\Types\Method;
use OneLang\One\Ast\Types\IMethodBase;

class InferTypesPlugin {
    public $main;
    public $errorMan;
    public $name;
    
    function __construct($name) {
        $this->name = $name;
        $this->errorMan = null;
    }
    
    function canTransform($expr) {
        return false;
    }
    
    function canDetectType($expr) {
        return false;
    }
    
    function transform($expr) {
        return $expr;
    }
    
    function detectType($expr) {
        return false;
    }
    
    function handleProperty($prop) {
        return false;
    }
    
    function handleLambda($lambda) {
        return false;
    }
    
    function handleMethod($method) {
        return false;
    }
    
    function handleStatement($stmt) {
        return false;
    }
}
